### ISSUE TITLE
Map::queryPointAnnotations returns empty

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#8799]
`Map::queryPointAnnotations` occasionally returns a 0 size when point annotations are found in the current viewport. For MarkerViews this results in flickering as they are removed/readded based on the results coming from Map::queryPointAnnotations.

![ezgif com-video-to-gif 42](https://cloud.githubusercontent.com/assets/2151639/25326558/01c5da76-28d1-11e7-990b-8d854f3854ef.gif)

Example output from gif above:

> E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 0
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 0
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 0
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2
 E/MarkerViewManager: MarkerViews in rect 2

This issue is a follow up from https://github.com/mapbox/mapbox-gl-native/issues/8367 and it's possible that the root problem could be related to https://github.com/mapbox/mapbox-gl-native/issues/6762.

**Comment by tobrun** [comment]
I replaced the MarkerView with Marker (PointAnnotation). While the marker itself doesn't flicker, I'm seeing similar output when it comes to `Map::queryPointAnnotations`.

```
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 0
E/AnnotationManager: Markers in rect 1
E/AnnotationManager: Markers in rect 0
```
I'm thinking that this could be related to the asynchronous behaviour of adding markers (a marker will only be available to query until it has been rendered) and that updating the position shows the same issue.

**Comment by ChrisLoer** [comment]
@tobrun How easy is this to reproduce? It would be good to test now that the fix for #10778 has gone into `master`. This issue was reported before we introduced the problem in #10778, but the code has changed enough we may have incidentally fixed the problem. If not, I can investigate.

**Comment by tobrun** [comment]
quick test didn't show the issue. Closing!
